### PR TITLE
Implement Natural Number Objects, and prove some properties

### PR DESCRIPTION
**Pull request by TOTBWF**
This PR implements [natural number objects](https://ncatlab.org/nlab/show/natural+numbers+object), and also shows that they are Initial F-Algebras for the Maybe Endofunctor.

### CODE REVIEW

File: src/Categories/Object/NaturalNumber.agda [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+open Equiv
+
+open Terminal 𝒞-Terminal
+
+private
+  variable
+    A B C D X Y Z : Obj
+    h i j : A ⇒ B
+
+record NaturalNumber : Set (o ⊔ ℓ ⊔ e) where
```
Comment (HuStmpHrrr) [inline src/Categories/Object/NaturalNumber.agda:24]: not sure if would make sense to provide a predicate version, like `IsNaturalNumber`?

Comment (TOTBWF) [reply src/Categories/Object/NaturalNumber.agda:24]: Good point! We should probably do this for all of our `Object`s moving forward 🙂 Perhaps we should add something to the design doc about this?

Comment (HuStmpHrrr) [reply src/Categories/Object/NaturalNumber.agda:24]: let's put some thought into it. in fact, all definitions of objects and diagrams are in predicate style.

Comment (TOTBWF) [reply src/Categories/Object/NaturalNumber.agda:24]: I can think of a few off the top of my head that aren't, but definitely should be (`Pullback` and `Coequalizer` are ones that comes to mind immediately)

Comment (HuStmpHrrr) [reply src/Categories/Object/NaturalNumber.agda:24]: yeah probably `Terminal` is not. `Product` is though. sort of inconsistent indeed.

**Comment by JacquesCarette** [comment]
So should I wait for further commits, or is this ready as is, and will be augmented later?

**Comment by TOTBWF** [comment]
Sorry, life has been getting in the way of Agda recently! I'll add a predicate form of this, and then we ought to be good to go.

**Comment by JacquesCarette** [comment]
I completely understand - my backlog of stuff on here shows the same pattern!

**Comment by TOTBWF** [comment]
Finally got around to tidying this up! Should be good to merge.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**